_episodes/04-Data visualisation with ggplot2.md

```diff
@@ -3,299 +3,361 @@ title: "Data visualisation with ggplot2"
+considered to be global settings. In other words, they will be considered the 
+'default' settings for any geom layer that you add.
+
+As was mentioned earier, the `aes` function is flexible in terms of which layer 
+it is defined in. This means that you can specify the mapping for a particular 
+geom independently rather than as a global setting.
 
-- Anything you put in the `ggplot()` function can be seen by any geom layers
-  that you add (i.e., these are universal plot settings). This includes the x- and
-  y-axis mapping you set up in `aes()`.
-- You can also specify mappings for a given geom independently of the
-  mappings defined globally in the `ggplot()` function. So the following works OK.
-  
 ~~~
-ggplot(data = SAFI_results) +
-  aes(x = A11_years_farm, y = B16_years_liv) +
-  geom_point()
+ggplot(data = SAFI_results) + 
+      geom_point(aes(x = A11_years_farm, y = B16_years_liv))
 ~~~
 
-- The `+` sign used to add layers must be placed at the end of each line containing
-a layer. If, instead, the `+` sign is added in the line before the other layer,
-`ggplot2` will not add the new layer and will return an error message.
-
+Note: Often, deciding whether to include mapping as a global setting or 
+independently within geoms is a personal preference or habit. However, if, for 
+example, you are adding several geom layers using the same mapping settings, 
+including `aes` in the `ggplot` function will save you time and minimize the risk 
+of typos.
 
-## Building your plots iteratively
+## Building Plots Iteratively
 
-Building plots with ggplot is typically an iterative process. We start by
-defining the dataset we'll use, lay out the axes, and choose a geom:
+Building plots with `ggplot2` is typically an iterative process. We start by
+defining the dataset we want to use, identifing our mapping aesthetics, and 
+choosing a geom.
 
 ~~~
-
 ggplot(data = SAFI_results, aes(x = A11_years_farm, y = B16_years_liv)) +
-  geom_point()
-  
+  geom_point() 
 ~~~
 
-Then, we start modifying this plot to extract more information from it. For
-instance, we can add transparency (`alpha`) to avoid overplotting:
+Next, we may want to start modifying this plot to make it more visualy appealing 
+or include more information. For instance, we can define the transparency of our 
+points using `alpha` to avoid overplotting.
 
 ~~~
-
 ggplot(data = SAFI_results, aes(x = A11_years_farm, y = B16_years_liv)) +
   geom_point(alpha = 0.1)
-  
 ~~~
 
-We can also add colors for all the points:
+We can decide that we also prefer all of the points to be blue.
 
 ~~~
 ggplot(data = SAFI_results, aes(x = A11_years_farm, y = B16_years_liv)) +
     geom_point(alpha = 0.1, color = "blue")
 ~~~
 
-Or to color each species in the plot differently:
+Upon further investigation, perhaps we decide that each point should be a 
+different colour, based on which species it represents - information that is 
+found in the column called `species_id`.
```

Comment:
is this dataset also using `species_id` or is this coming from the ecology lesson?

Reply:
I'm unsure - being unfamiliar with the dataset, I just looked to standardize the code that was alreayd there - in some places, species_id and weight were used so I continued to use those when I updated other pieces (e.x. needed to use a continuous and a categorical variable so used these since they were already mentioned in the lesson) - I'm happy to go through the code and datasets in more detail and sort out exactly which variables will be used / available :)

Reply:
I had a quick look at the dataset, and it does seem that `species_id` and `weight` are leftovers from copying from the ecology lesson as I don't see them in the data set used here.

Reply:
Ok! I'll take a look and figure out what variables might be appropriate - is it better to do that as a separate PR?

Reply:
It's better to keep updating this PR at this stage. Thanks!